kura/examples/org.eclipse.kura.example.eddystone.scanner/src/main/java/org/eclipse/kura/example/eddystone/scanner/EddystoneScannerExample.java

```diff
@@ -0,0 +1,261 @@
+
+        if (this.cloudClient != null) {
+            this.cloudClient.release();
+        }
+
+        logger.debug("Deactivating Eddystone Scanner Example... Done.");
+    }
+
+    protected void updated(Map<String, Object> properties) {
+        logger.debug("Updating Eddystone Scanner Example...");
+
+        releaseResources();
+
+        if (this.handle != null) {
+            this.handle.cancel(true);
+        }
+
+        if (this.worker != null) {
+            this.worker.shutdown();
+        }
+
+        doUpdate(properties);
+
+        logger.debug("Updating Eddystone Scanner Example... Done");
+    }
+
+    private void doUpdate(Map<String, Object> properties) {
+        for (Entry<String, Object> entry : properties.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            if (key.equals(PROPERTY_INAME)) {
+                this.adapterName = (String) value;
+            } else if (key.equals(PROPERTY_TOPIC_PREFIX)) {
+                this.topicPrefix = (String) value;
+            } else if (key.equals(PROPERTY_RATE_LIMIT)) {
+                this.rateLimit = (Integer) value;
+            } else if (key.equals(PROPERTY_ENABLE)) {
+                this.enableScanning = (Boolean) value;
+            } else if (key.equals(PROPERTY_TIMEOUT)) {
+                this.timeout = (Integer) value;
+            }
+        }
+
+        if (this.enableScanning) {
+            this.worker = Executors.newSingleThreadExecutor();
+            this.handle = this.worker.submit(() -> setup());
+        }
+    }
+
+    private void setup() {
+        BluetoothLeAdapter bluetoothLeAdapter = this.bluetoothLeService.getAdapter(this.adapterName);
+        if (bluetoothLeAdapter != null) {
+            this.bluetoothLeEddystoneScanner = this.bluetoothLeEddystoneService.getBeaconScanner(bluetoothLeAdapter,
+                    this.bluetoothLeEddystoneCodec);
+            this.bluetoothLeEddystoneScanner.addBeaconListener(this);
+            try {
+                this.bluetoothLeEddystoneScanner.startBeaconScan(this.timeout * 1000L);
+            } catch (KuraException e) {
+                logger.error("iBeacon scanning failed", e);
+            }
+        }
+    }
+
+    private void releaseResources() {
+        if (this.bluetoothLeEddystoneScanner != null && this.bluetoothLeEddystoneScanner.isScanning()) {
+            try {
+                this.bluetoothLeEddystoneScanner.stopBeaconScan();
+            } catch (KuraException e) {
+                logger.error("iBeacon scan stop failed", e);
+            }
+        }
+    }
+
+    private double calculateDistance(int rssi, int txpower) {
+
+        double distance;
+
+        int ratioDB = txpower - rssi;
+        double ratioLinear = Math.pow(10, (double) ratioDB / 10);
+        distance = Math.sqrt(ratioLinear);
+
+        // See http://stackoverflow.com/questions/20416218/understanding-ibeacon-distancing/20434019#20434019
```

Comment:
No, it's only for reference. However, I removed the comment because is not relevant.